src/components/AddTask.jsx

```diff
@@ -0,0 +1,66 @@
+import React, { useState, useRef, useEffect } from 'react';
+import { ReactComponent as DeleteIcon } from 'assets/delete.svg';
+import { useDispatch } from 'react-redux';
+import { addTodo, setIsAddingTask } from 'redux/actions/TodoAction';
+import { KEY_ENTER, RESPONSE_ERROR } from 'utils/constants';
+import { validate } from 'utils/helpers/index';
+import Button from 'components/ui/Button';
+
+function AddTask() {
+  const [error, setError] = useState(null);
+  const [taskDetails, setTaskDetails] = useState('');
+  const dispatch = useDispatch();
+  const textAreaRef = useRef(null);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const validateDetails = validate(taskDetails);
+    if (validateDetails.status === RESPONSE_ERROR) {
+      setError(validateDetails.message);
+    } else {
+      setError(null);
+      dispatch(addTodo(validateDetails.text));
+      dispatch(setIsAddingTask(false));
+      setTaskDetails('');
+    }
```

Comment:
Could we use early return here ?
```suggestion
    if (validateDetails.status === RESPONSE_ERROR) {
      setError(validateDetails.message);
      
      return
    }
    
    setError(null);
    dispatch(addTodo(validateDetails.text));
    dispatch(setIsAddingTask(false));
    setTaskDetails('');
    
```